src/data/trip.js

```diff
@@ -35,22 +45,26 @@ export default function trip() {
       return tripRef.get();
     },
 
-    getAllTrips: tripRefs => {
-      return Promise.all(
-        tripRefs.map(tripRef => {
-          return tripRef.get();
-        }),
-      );
+    joinTrip: tripId => {
+      const batch = db.batch();
+      const userRef = db.collection('users').doc(currentUser.uid);
+      const tripRef = db.collection('trips').doc(tripId);
+
+      batch.update(tripRef, {
+        attendees: firebase.firestore.FieldValue.arrayUnion(userRef),
+      });
+      batch.update(userRef, {
+        trips: firebase.firestore.FieldValue.arrayUnion(tripRef),
+      });
```

Comment:
@yongbingao I updated your code to combine both the update on the trip and the user profile when joining a trip. This separates the logic from updating a regular user profile or trip details.